nvm.sh

```diff
@@ -1904,6 +1904,8 @@ nvm_install_source() {
   fi
 
   nvm_err "nvm: install ${VERSION} failed!"
+  # shellcheck disable=SC2031
```

Comment:
by my reading of https://github.com/koalaman/shellcheck/wiki/SC2031, this shellcheck error is actually very serious and should not be bypassed.

I think that in order for `TMPDIR` to be available outside of the `if` chain, we need to refactor that chain so that it's defined at the function level.